### PR TITLE
Multiply -> Product

### PR DESCRIPTION
**Pull request by farizrahman4u**
I am not 100 % sure of this either. But I think the right relationship is add : sum = multiply : product. Add is a verb, and sum is the result of addition; same goes for multiply / product. Since the "+" layer is called `Sum`  , `Product` would be more consistent than `Multiply` even though it was called `mul` mode in Keras 1.

### CODE REVIEW

**Comment by fchollet** [comment]
In part, it's meant to mirror the Numpy API terminology. `product` multiplies elements along an axis (or several), and `multiply` multiplies a number of arrays. https://docs.scipy.org/doc/numpy/reference/generated/numpy.multiply.html

In larger part, it's because we want verbs (actions). Merge is a verb, sum is a verb, concatenate is a verb, multiply is a verb. Product is not.

Following that reasoning, maybe `sum` could be renamed `add` (to mirror Numpy). Or maybe not.

**Comment by farizrahman4u** [comment]
Right. Changed sum to add. Might feel a bit unnatural at first, but its better this way than have 2 function with same name. (keras.backend.sum and keras.layers.sum). Also, when I do `from keras.layers import *` i don't want it to override python's built-in sum function.

File: keras/layers/merge.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def sum(inputs, **kwargs):
 
 
 def multiply(inputs, **kwargs):
-    """Functional interface to the `Multiply` layer.
+    """Functional interface to the `Product` layer.
```
Comment (fchollet) [inline keras/layers/merge.py]: Fix.

Comment (farizrahman4u) [inline keras/layers/merge.py]: Done.

File: keras/layers/merge.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 
-def sum(inputs, **kwargs):
+def add(inputs, **kwargs):
     """Functional interface to the `Sum` layer.
```
Comment (fchollet) [inline keras/layers/merge.py]: It still says "sum" here...

Comment (farizrahman4u) [inline keras/layers/merge.py]: fixed

**Comment by fchollet** [comment]
I incorrectly merged this; the `add` function was still calling the now non-existing `Sum` function. This was no visible from the GH PR interface, and since tests are supposed to fail at this point, I didn't even check them.

This PR has triggered 5 integration tests run on Travis and ended up being obviously mistaken. For a trivial renaming change. This is not normal...

**Comment by farizrahman4u** [comment]
Screwed up. Both "Sum" and "Add" maps to the same word sense in my head. Apologies.
